portfolio/src/main/webapp/script.js

```diff
@@ -29,3 +29,45 @@ function showSlides(n) {
   }
   slides[slideIndex-1].style.display = "block";
 }
+
+/**
+ * Fetches a message from the server and adds it to the page.
+ */
+function getMessage() {
+  console.log('Fetching the message');
+
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
+ */
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addMsgToDom() function.
+  textPromise.then(addMsgToDom);
+}
+
+/** Adds message to the DOM. */
+function addMsgToDom(message) {
+  console.log('Adding message to dom: ' + message);
```

Comment:
Not sure if we should keep all of the logging messages. If you'd like to keep them for now, I guess we can take them out in a more final version of the portfolio.